lib/cheffish/node_properties.rb

```diff
@@ -0,0 +1,107 @@
+require 'chef_compat/mixin/properties'
+
+module Cheffish
+  module NodeProperties
+    include ChefCompat::Mixin::Properties
+
+    # Grab environment from with_environment
+    def initialize(*args)
+      super
+      chef_environment run_context.cheffish.current_environment
+    end
+
+    property :name, Cheffish::NAME_REGEX, name_property: true
+    property :chef_environment, Cheffish::NAME_REGEX
+    property :run_list, Array # We should let them specify it as a series of parameters too
+    property :attributes, Hash
+
+    # attribute 'ip_address', '127.0.0.1'
+    # attribute [ 'pushy', 'port' ], '9000'
+    # attribute 'ip_addresses' do |existing_value|
+    #   (existing_value || []) + [ '127.0.0.1' ]
+    # end
+    # attribute 'ip_address', :delete
+    attr_accessor :attribute_modifiers
+    def attribute(attribute_path, value=Chef::NOT_PASSED, &block)
```

Comment:
To depend on `Chef::NOT_PASSED` it seems like we need to bump the [chef dependency](https://github.com/chef/cheffish/blob/b600412db1beced66e2c65c38af5a01663bb66a4/cheffish.gemspec#L18) up to `~> 12.5` (or whenever `NOT_PASSED` was introduced to chef)